client/src/pages/NrMap.js

```diff
@@ -33,61 +33,75 @@ const Cards = styled.div`
   margin: 10px;
 `;
 
-class NrMap extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      latitude: 52.520008,
-      longitude: 13.404954,
-      rec: [],
-      plcDtl: [],
-    };
-    this.googleMapRef = React.createRef();
-  }
-  componentDidMount() {
+const map = {
+  googleMap: null,
+  marker: null,
+  place: null,
+};
+
+const NrMap = () => {
+  const [coordinates, setCoordinates] = useState({
+    latitude: 52.520008,
+    longitude: 13.404954,
+  });
+  // rec-> list of hospitals according to nearbySearch
+  const [rec, setRec] = useState([]);
+  // plcDtl-> list of hospitals with requested details
+  const [plcDtl, setPlcDtl] = useState(() => []);
+
+  const googleMapRef = useRef();
+
+  useEffect(() => {
     const googleMapScript = document.createElement("script");
     googleMapScript.src = url;
     window.document.body.appendChild(googleMapScript);
 
     googleMapScript.addEventListener("load", () => {
-      this.googleMap = this.createGoogleMap();
-      this.marker = this.createMarker();
-      this.place = this.places();
+      map.googleMap = createGoogleMap();
```

Comment:
I'm not sure if it's good practice in React to update state like this, since `map` is just an object right now and not part of React state.

Reply:
I agree. I think that the React way would be by creating a ref with userRef() since in functional components, differently from class components, they are meant to store anything (not just dom nodes refs). Personally I still find it a bit confusing, maybe because I still thing in refs as a dom refs. Anyway I've have changed it, and removed the global map variable and it looks more React :D

Reply:
Yes, I agree. I think that the react way is by useRef(). I've changed it to a ref. Also got rid of the other two variables since they are not storing anything (createMarker and placeDetails don't return anything).

Reply:
Good to know... I remember sometimes there can be problems integrating other libraries with React because of the lifecycle hooks, didn't know you could use refs for things other than DOM elements... I don't really find this in the docs though
https://reactjs.org/docs/integrating-with-other-libraries.html
https://reactjs.org/docs/refs-and-the-dom.html#when-to-use-refs

Reply:
it seems that the perspective has changed because of hooks. it's in the useRef hooks docs https://reactjs.org/docs/hooks-reference.html#useref